modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArtifactDeployer.java

```diff
@@ -29,9 +36,16 @@ public void deploy(T artifact, boolean startArtifact) {
     try {
       artifact.install();
       doInit(artifact);
-      if (startArtifact) {
+      if (startArtifact && shouldStartArtifact(artifact)) {
         artifact.start();
       }
+
+      ArtifactStoppedDeploymentListener artifactStoppedDeploymentListener =
```

Comment:
use the interface (`ArtifactStoppedListener`) in the var declaration, not the impl